src/components/SearchInput.tsx

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React, { useState, useEffect, memo, useCallback } from 'react'
+import { useTranslation } from 'react-i18next'
+import Search from '@assets/images/search.svg'
+import Close from '@assets/images/closeModal.svg'
+import { useDebouncedCallback } from 'use-debounce/lib'
+import { BoxProps } from '@shopify/restyle'
+import Box from './Box'
+import TextInput from './TextInput'
+import TouchableOpacityBox from './TouchableOpacityBox'
+import { Theme } from '../theme/theme'
+
+export const HotspotFilterKeys = ['my_hotspots', 'all_hotspots'] as const
+export type HotspotFilterType = typeof HotspotFilterKeys[number]
+
+type Props = BoxProps<Theme> & {
+  onSearch: (searchTerm: string) => void
+  initialValue?: string
+}
+const HotspotSearch = ({ onSearch, initialValue = '', ...boxProps }: Props) => {
+  const { t } = useTranslation()
+  const [searchTerm, setSearchTerm] = useState(initialValue)
+
+  useEffect(() => setSearchTerm(initialValue), [initialValue])
+
+  const debouncedSearchTermHandler = useDebouncedCallback((term: string) => {
+    onSearch(term)
```

Comment:
Should hotspot names with spaces also work? It seems to only work if I type dashes?

Reply:
Another thing, when I am on the "My Hotspots" tab, the hint text made me feel like I could also search a location to filter my own hotspots. Like if I type city names I would see my hotspots in that city.

Reply:
oh, ok, I should be able to add that.